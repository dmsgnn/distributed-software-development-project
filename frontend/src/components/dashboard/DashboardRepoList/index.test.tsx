import { axe } from 'jest-axe';

import { act, render } from 'test-utils';
import { DashboardRepoList } from './';

const mockRouterPush = jest.fn();
jest.mock('next/router', () => ({ useRouter: jest.fn(() => ({ push: mockRouterPush })) }));

describe('DashboardRepoList', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<DashboardRepoList />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<DashboardRepoList />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});
});
