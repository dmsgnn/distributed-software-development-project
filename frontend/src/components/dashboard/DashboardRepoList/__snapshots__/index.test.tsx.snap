// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRepoList should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-bl1qof"
  >
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
  </div>
</DocumentFragment>
`;
